### PR TITLE
Update InputStreamPumper.java

### PR DESCRIPTION
**Pull request by saowu**
## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java [hunk 1]
```diff
@@ -87,6 +88,12 @@ public static void transferTo(InputStream in, Writable out) throws IOException {
     return CompletableFuture.runAsync(() -> {
       try {
         InputStreamPumper.transferTo(in, out);
+      } catch (InterruptedIOException e){
+        if (!Thread.currentThread().isInterrupted()) {
```
Comment (rohanKanojia) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java:92]: Apologies for another comment, you need to remove this same `if-else` block from `catch (Exception e)` block

Comment (saowu) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java:92]: In fact, I don't quite understand what you mean. You can modify it directly

Comment (saowu) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java:92]: Is this correct?

**Review by rohanKanojia — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java [hunk 2]
```diff
@@ -87,12 +88,14 @@ public static void transferTo(InputStream in, Writable out) throws IOException {
     return CompletableFuture.runAsync(() -> {
       try {
         InputStreamPumper.transferTo(in, out);
-      } catch (Exception e) {
+      } catch (InterruptedIOException e){
         if (!Thread.currentThread().isInterrupted()) {
```
Comment (shawkins) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java:92]: We don't really care about this distinction here.  It should just be:

```
      } catch (InterruptedIOException e){
        LOGGER.debug("Interrupted while pumping stream.");
      } catch (Exception e) {
        if (!Thread.currentThread().isInterrupted()) {
          LOGGER.error("Error while pumping stream.", e);
        } else {
          LOGGER.debug("Interrupted while pumping stream.");
        }
      }
```
That is assuming there is something happening where an InterruptedException is then causing an unchecked exception or a non- InterruptedIOException.

**Review by shawkins — APPROVED** [review]
LGTM

**Pull request merged**